velox/functions/sparksql/Map.cpp

```diff
@@ -130,26 +105,12 @@ class MapFunction : public exec::VectorFunction {
     // Setting keys and value elements
     auto keysResult = mapResult->mapKeys();
     auto valuesResult = mapResult->mapValues();
-    keysResult->resize(rows.end() * mapSize);
-    valuesResult->resize(rows.end() * mapSize);
-
-    VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
-        setKeysResultTyped,
-        keyType->kind(),
-        mapSize,
-        args,
-        keysResult,
-        context,
-        rows);
+    const auto resultSize = rows.end() * mapSize;
```

Comment:
This is existing code, but I think there is a bug here. keysResult and valuesResult may already contain values that need to be preserved and this code will overwrite these. 

I'll merge this PR as is, but, please, run fuzzer and fix whatever bugs it discovers.

```
velox/expression/tests/spark_expression_fuzzer_test --logtostderr --only map,switch --special_forms switch --velox_fuzzer_enable_complex_types --duration_sec 60
```

CC: @rui-mo

Reply:
For example, I'm seeing this failure:

```
velox/expression/tests/spark_expression_fuzzer_test --logtostderr --only map,if --special_forms if --velox_fuzzer_enable_complex_types --duration_sec 60 -

I20231123 01:31:36.178881 10397484 ExpressionFuzzer.cpp:1313] ==============================> Started iteration 902 (seed: 3614801884)
I20231123 01:31:36.187767 10397484 ExpressionVerifier.cpp:86] Executing expression 0 : if(if("c0","c1",null),19 elements starting at 546 {13 elements starting at 5169 {[5169->507] [507->3783] [3783->4406] false, [5170->2311] [2311->3130] [3130->3517] false, [5171->3031] [3031->1168] [1168->3757] true, [5172->5692] [5692->2053] [2053->2646] true, [5173->5638] [5638->5490] [5490->5507] false, ...} => [546->375] 1873667577, 10 elements starting at 5182 {[5182->4078] [4078->1850] [1850->2405] true, [5183->1252] [1252->2713] [2713->2121] false, [5184->158] [158->5885] [5885->3448] true, [5185->1198] [1198->4863] [4863->5868] true, [5186->2188] [2188->1360] [1360->1347] true, ...} => [547->383] 1241401043, 13 elements starting at 5192 {[5192->4870] [4870->4039] [4039->2402] true, [5193->5005] [5005->3819] [3819->1266] true, [5194->1533] [1533->5268] [5268->9] true, [5195->2025] [2025->2598] [2598->134] true, [5196->5485] [5485->4286] [4286->5651] true, ...} => [548->114] 1560512028, 4 elements starting at 5205 {[5205->5751] [5751->3194] [3194->4170] false, [5206->2230] [2230->3886] [3886->3695] false, [5207->1656] [1656->28] [28->176] false, [5208->2331] [2331->1352] [1352->3906] true} => [549->210] 904276463, 17 elements starting at 5209 {[5209->3680] [3680->2745] [2745->5573] true, [5210->2138] [2138->82] [82->4468] false, [5211->5554] [5554->180] [180->3354] true, [5212->4653] [4653->2033] [2033->291] true, [5213->2708] [2708->4064] [4064->1922] true, ...} => null, ...},map("c2",if(if(true,"c3"),"c4")))
E20231123 01:31:36.191498 10397484 Exceptions.h:69] Line: /Users/mbasmanova/cpp/velox-1/velox/vector/ComplexVector.cpp:725, Function:validateArrayVectorBase, Expression: rawOffsets_[i] + rawSizes_[i] - 1 < minChildVectorSize (113 vs. 100) ArrayVectorBase must only point to indices within the base vector's size. Index: 44., Source: RUNTIME, ErrorCode: INVALID_STATE
E20231123 01:31:36.192513 10397484 ExpressionVerifier.cpp:179] Common eval: Exceptions other than VeloxUserError are not allowed.
I20231123 01:31:36.192560 10397484 ExpressionVerifier.cpp:258] Skipping persistence because repro path is empty.
libc++abi: terminating due to uncaught exception of type facebook::velox::VeloxRuntimeError: Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: (113 vs. 100) ArrayVectorBase must only point to indices within the base vector's size. Index: 44.
Retriable: False
Expression: rawOffsets_[i] + rawSizes_[i] - 1 < minChildVectorSize
Context: map(c2, switch(switch(true:BOOLEAN, c3), c4))
Top-Level Context: switch(switch(c0, c1, null:BOOLEAN), 19 elements starting at 0 {13 elements starting at 0 {false, false, true, true, false, ...} => 1873667577, 10 elements starting at 13 {true, false, true, true, true, ...} => 1241401043, 13 elements starting at 23 {true, true, true, true, true, ...} => 1560512028, 4 elements starting at 36 {false, false, false, true} => 904276463, 17 elements starting at 40 {true, false, true, true, true, ...} => null, ...}:MAP<ARRAY<BOOLEAN>,INTEGER>, map(c2, switch(switch(true:BOOLEAN, c3), c4)))
Function: validateArrayVectorBase
File: /Users/mbasmanova/cpp/velox-1/velox/vector/ComplexVector.cpp
Line: 725
```

Reply:
Sure, let me check.

Reply:
Here is the PR to fix this issue, thanks.
https://github.com/facebookincubator/velox/pull/7727